src/Data/Tree/AVL/Store/Pure.hs

```diff
@@ -38,20 +38,40 @@ type StoreT h k v = ReaderT (TVar (State h k v))
 -- | Nicer way to assign things via lenses.
 asState :: MonadIO m => StateT (State h k v) m a -> StoreT h k v m a
 asState action = do
-    var <- ask
-    st  <- liftIO $ atomically $ readTVar var
-    (b, st') <- lift $ runStateT action st
-    liftIO $ atomically $ writeTVar var st'
+    var      <- ask
+    st       <- liftIO $ atomically $ readTVar  var
+    (b, st') <- lift   $ runStateT action st
+    _        <- liftIO $ atomically $ writeTVar var st'
```

Comment:
I know very well that you're a fan of vertical alignment, but let's just consider another opinion: http://www.zeyalabs.ch/posts/2013/vertical-alignment-sucks/

Reply:
Readability of code is of higher concern for me, so that one I'll not undo.